doc/source/advanced/datatype.rst

```diff
@@ -1,31 +1,44 @@
 ***********************************
-Data Types in Lux
+Defining Data Types
 ***********************************
 
 .. note:: You can follow along this tutorial in a Jupyter notebook. 
 
-Currently, Lux supports four data types:
+.. contents:: :local:
+
+
+Data Types
+==========
+
+Data types in Lux are used to generate suggested visualizations. 
+These data types are automatically inferred based on the input `dataframe` which are then used by the
+`Compiler <https://lux-api.readthedocs.io/en/latest/source/advanced/executor.html>`_ to fill in the missing information in each `Clause`. 
+While `Clauses` are not always specified, the system infers intent based on user inputs as well as the structure of the `dataframe`.
+These `Clauses` determine which visualizations will be displayed.
+
+.. note:: For more information about how Lux creates visualizations from a `frame`, see `this page <https://lux-api.readthedocs.io/en/latest/source/advanced/architecture.html>`_.
+
+Note that these data types are different from Pandas `dtype`. 
 
-* Quantitative
-* Temporal
-* Nominal
-* ID
 
-Note that these data types are different from Pandas `dtype`. These data types are automatically inferred based on the input `dataframe`.
 The following code snippet shows what data types were inferred:
 
 .. code-block:: python
 
   df = pd.read_csv("https://github.com/lux-org/lux-datasets/blob/master/data/college.csv?raw=true")
   df.data_type
 
+.. image:: https://github.com/jinimukh/lux-resources/blob/datatype/doc_img/datatype-4.png?raw=true
+  :width: 300
+  :align: center
 
 Quantitative
 ------------
-The quantitative data type is used when there is a count or measure of a certain attribute. 
+Quantitative data is used to describe numerical data. 
+The is usually used when Lux detects that there is a count or measure of a certain attribute. 
```

Comment:
Small typo at the start of this line, maybe adjust it to read something like:
"This data type is typically assigned when Lux detects that there is a count or measure of a certain attribute."